sormas-stats-sitrep/utils/caseExport.R

```diff
@@ -0,0 +1,190 @@
+caseExport <- function(sormas_db, fromDate, toDate){
+  # This function creates two tables as output.
+  # The first table is a line listing of all cases reported in the defined
+  # time period. This line listing of cases includes all the variables as columns
+  # that are necessary for the creation of the SitRep, apart from population and 
+  # geo shapes data.
+  # The second table contains the population and geo data.
+  # It contains data on the population by agegroup and gender in each district.
+  # The two tables are output separately, because each case corresponds to one 
+  # district, but each district corredponds to multiple population counts
+  # (by agegroup and gender). Therefore a line listing by case would be
+  # impractical and also not very suitable for the computation of the 
+  # subsequent parts of the situation report.
+  
+  ## Computing default time based on 365 days in the past if not provided by user
+  if(missing(fromDate) | missing(toDate)){
+    fromDate = as.character(Sys.Date() - delay) 
+    toDate = as.character(Sys.Date()) 
+  }
+
+  ## Loading tables from SORMAS DB
+  
+  # Case table SQL query
+  # (added vaccination status, because it could be interesting for a 
+  # possible extension of the SitRep)
+  queryCases <- paste0("SELECT  DISTINCT id AS case_id,
+    district_id,
+    region_id,
+    caseclassification AS case_classification,
+    EndOfIsolationReason AS case_end_of_iso_reason,
+    reportdate AS report_date_case,
+    disease,
+    vaccinationstatus AS vaccination_status, 
+    symptoms_id,
+    person_id,
+    hospitalization_id 
+  
+    FROM public.cases 
+    
+    WHERE deleted = FALSE 
+    and caseclassification != 'NO_CASE'
+    and reportdate between '", fromDate, "' and '", toDate, "' 
+    and disease = 'CORONAVIRUS' 
+                       ")
+  cases <- dbGetQuery(sormas_db,queryCases)
+  
+  
+  # Person table SQL query
+  queryPersons <- paste0("SELECT DISTINCT id AS person_id,
+    approximateAge AS age_person,
+    sex AS sex_person,
+    deathDate AS death_date_person,
+    causeOfDeathDisease AS cause_of_death_disease_person
+
+    FROM public.person
+
+    WHERE id IN (SELECT person_id FROM public.cases
+      WHERE deleted = FALSE 
+      and caseclassification != 'NO_CASE'
+      and reportdate between '", fromDate, "' and '", toDate, "' 
+      and disease = 'CORONAVIRUS')
+                         ")
+  # Only persons with a person_id that also appears in a case 
+  # are queried from the database
+  persons <- dbGetQuery(sormas_db,queryPersons)
+  
+  # Symptoms table SQL query
+  querySymptoms <- paste0("SELECT DISTINCT id AS symptoms_id,
+    onsetDate AS onset_date
+    
+    FROM public.symptoms
+    
+    WHERE id IN (SELECT symptoms_id FROM public.cases
```

Comment:
the same comment since cases has been selected already, sample code:
  #loading symptom data corresponding to selected cases only
  querySymptom = sprintf(
    "SELECT id AS symptom_id, onsetdate AS onset_date
     FROM public.symptoms
     WHERE id in (%s)", paste("'", base::unique(c(case$symptoms_id)), "'",collapse=",") 
  )
  symptoms =  dbGetQuery(sormas_db,querySymptom)